sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -74,6 +74,24 @@ gnrc_netif_t *gnrc_netif_create(char *stack, int stacksize, char priority,
     return netif;
 }
 
+bool gnrc_netif_dev_is_6lo(const gnrc_netif_t *netif)
+{
+    switch (netif->device_type) {
+#ifdef MODULE_GNRC_SIXLOENC
+        case NETDEV_TYPE_ETHERNET:
+            return (netif->flags & GNRC_NETIF_FLAGS_6LO);
+#endif
```

Comment:
In theory this can also use link-time optimization as well:

```C
        case NETDEV_TYPE_ETHERNET:
            if (IS_USED(MODULE_GNRC_SIXLOENC)) {
                return (netif->flags & GNRC_NETIF_FLAGS_6LO);
            }
            /* intentionally falls through */
```